app/customer.py

```diff
@@ -0,0 +1,46 @@
+from app.car import Car
+
+
+class Customer:
+    def __init__(self,
+                 name: str,
+                 product_cart: dict[str:int],
+                 location: list[int],
+                 money: int,
+                 car: Car) -> None:
+
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+
+        self.__home_location = location
+        self.__money_to_spend_in_shop: dict[str:float] = {}
+
+    def __repr__(self) -> str:
+        return f"{self.name}"
+
+    def go_to_home(self) -> float:
+        if self.location == self.__home_location:
+            raise RuntimeError(f"Customer {self.name} is already at home!")
+
+        return self.go_to_location(self.__home_location)
+
+    def go_to_location(self, location: list[int]) -> float:
+        distance = calculate_distance(self.location, location)
+        self.location = location
+        return distance
```

Comment:
The `go_to_location` method calculates the distance but does not account for the fuel cost using the `Car` class. Consider using the `calculate_fuel_cost` method from the `Car` class to deduct the fuel cost from the customer's money.